daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -2402,8 +2402,9 @@ case class DFDLInfoset(di: Node, parent: Infoset) {
     // The expected infoset is loaded using the normalizeCRLFtoLF mode (which
     // is the default), so MS-DOS/Windows CRLFs in expected data XML files will
     // have been converted into LF at this point.
-    if (elem ne null)
+    if ((elem ne null) && elem.toString.contains("\r")) {
       Assert.invariant(!elem.toString.contains("\r"))
```

Comment:
Thoughts on if we want to keep this assertion? This is going to convert every epected infoset to a string just to make sure there's no \r. Feels like that might be somewhat expensive?

Reply:
Once I get this debugged I will remove this one. Yes it is way too costly.